File: 03_intent/app/src/main/java/com/bomnie/a03_intent/MainActivity.java
package com.bomnie.a03_intent;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), MenuActivity.class);
                startActivityForResult(intent, 101);
            }
        });
    }

    // 메뉴 화면에서의 응답을 받아주는 콜백함수

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if(requestCode == 101){ // 메뉴 화면에서 전달받은 요청이 맞는가?
            String name = data.getStringExtra("name");
            Toast.makeText(getApplicationContext(), "메뉴 화면으로부터 응답 : "+name, Toast.LENGTH_SHORT).show();
        }
    }
}

File: HelloKotlin/app/src/main/java/com/bomnie/hellokotlin/MainActivity.kt
package com.bomnie.hellokotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() { // 상속은 :로 표시

    override fun onCreate(savedInstanceState: Bundle?) { // 변수명: 자료형
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun clickBtn(view: View) {
        // XML 에서 지정한 id를 곧바로 변수명으로 사용함
        tv.setText("Nice to meet you Kotlin!")
    }

    //onResume 메소드 오버라이드
    override fun onResume(){
        super.onResume()
        Toast.makeText(this, "onResume", Toast.LENGTH_SHORT).show()
    }
}
